service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java

```diff
@@ -99,64 +96,132 @@ public void setRestTemplate(RestTemplate restTemplate) {
+                                      @ApiParam(value = RemremGenerateServiceConstants.LenientValidation) @RequestParam(value = "okToLeaveOutInvalidOptionalFields", required = false, defaultValue = "false")  final Boolean okToLeaveOutInvalidOptionalFields,
+                                      @ApiParam(value = "JSON message", required = true) @RequestBody String body){
+        try {
+            JsonElement bodyJson = JsonParser.parseString(body);
+
+            return generate(msgProtocol, msgType, failIfMultipleFound, failIfNoneFound, lookupInExternalERs,
+                    lookupLimit, okToLeaveOutInvalidOptionalFields, bodyJson);
+
+        } catch (JsonSyntaxException e) {
+            JsonObject errorResponse = new JsonObject();
+            log.error("Unexpected exception caught due to parse json data", e.getMessage());
+            String exceptionMessage = e.getMessage();
+            errorResponse.addProperty("Status code", HttpStatus.BAD_REQUEST.value());
+            errorResponse.addProperty("result", "Fatal");
+            errorResponse.addProperty("message", "Invalid JSON parse data format due to: " + exceptionMessage);
+            return new ResponseEntity<>(errorResponse, HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
 
+    public ResponseEntity<?> generate(final String msgProtocol, final String msgType, final Boolean failIfMultipleFound, final Boolean failIfNoneFound, final Boolean lookupInExternalERs, final int lookupLimit, final Boolean okToLeaveOutInvalidOptionalFields, JsonElement bodyJson) {
+
+        JsonArray results = new JsonArray();
+        JsonObject errorResponse = new JsonObject();
         try {
-            bodyJson = erLookup(bodyJson, failIfMultipleFound, failIfNoneFound, lookupInExternalERs, lookupLimit);
+
+            if (bodyJson.isJsonArray()) {
+
+                JsonArray jsonArray = bodyJson.getAsJsonArray();
+                for (JsonElement element : jsonArray) {
+                    results.add(processEvent(element.getAsJsonObject(), msgProtocol, msgType,
+                            failIfMultipleFound, failIfNoneFound, lookupInExternalERs, lookupLimit,
+                            okToLeaveOutInvalidOptionalFields));
+                }
+                return new ResponseEntity<>(results, HttpStatus.OK);
+
+            } else if (bodyJson.isJsonObject()) {
+                JsonObject jsonObject = bodyJson.getAsJsonObject();
+                JsonObject jsonObject1 = processEvent(jsonObject, msgProtocol, msgType, failIfMultipleFound, failIfNoneFound,
+                        lookupInExternalERs, lookupLimit, okToLeaveOutInvalidOptionalFields);
+                return new ResponseEntity<>(jsonObject1, HttpStatus.OK);
+            } else {
+                errorResponse.addProperty("Status code", HttpStatus.BAD_REQUEST.value());
+                errorResponse.addProperty("result", "fail");
+                errorResponse.addProperty("message", "Invalid JSON format");
```

Comment:
Add more hints: something like "either single template or array of templates is expected".

Reply:
done